### PR TITLE
Fixes Elemental Converter and equip swap

### LINKED ISSUE #5907: Elemental converter buff should NOT be removed when Equip/Unequip 2handed weapons/shields
* **rAthena Hash**: Latest

* **Client Date**: 20200401

* **Server Mode**: Pre-Renewal (not sure if Renewal had its behavior as well so...)

* **Description of Issue**: 
  * Result: Elemental Converter buff is being removed when switching weapons/shields;
  * Expected Result: The buff should not be removed when equip/unequip weapons/shields; 
  * Note: I imagine the buff should be gone if you unequip the weapon without equipping a shield.
  * How to Reproduce: 
  * 1. Create any character,get some Elemental Converters(ID: 12117 for example), a weapon based on the class you create and a shield (Valkyrja's Shield ID: 2115 for example);
  * I used a Lord Knight with Claymore/Valk Shield and a Assassin Cross with Katar/Buckler; Also put some few stats points (50str, some dex.. )
  * 2. Summon a Phen monster (ID: 1158);
  * 3. Bare handed, attack Phen; 
  * 4. Use the elemental converter and then attack again; (Damage will almost double due to element effectiveness)
  * 5. Then equip weapon. The elemental converter buff still remain;
  * 6. Then equip the shield (It'll unequip the 2h weapon and equip the shield);
  * 7. Elemental converter buff will be gone!
  * Official Information: The profs provided here were obtained from a discursion on OriginRO discord (Thanks to Xellie). The video below is a supposedly test on Aegis. (Would be nice if anyone with Aegis access could confirm)
  

* **Modifications that may affect results**: None

https://user-images.githubusercontent.com/16859064/116487788-245b6780-a867-11eb-8e85-0a00147c79f2.mp4

### PR DESCRIPTION
**Pull request by aleos89**
* **Addressed Issue(s)**: Fixes #5907

* **Server Mode**: Pre-renewal and Renewal

* **Description of Pull Request**: 
  * Fixes Elemental Converter getting removed when swapping weapons/shields.
Thanks to @Tydus1!

### CODE REVIEW

File: src/map/skill.cpp [hunk 1]
```diff
@@ -18367,7 +18367,8 @@ void skill_enchant_elemental_end(struct block_list *bl, int type)
 	if (!sc->count)
 		return;
 
-	status_change_end(bl, SC_ENCHANTARMS, INVALID_TIMER); // Should always end
+	if (type != SC_NONE)
```
Comment (Lemongrass3110) [inline src/map/skill.cpp:18370]:
```suggestion
	// If it is not on equip change
	if (type != SC_NONE)
```

File: src/map/skill.cpp [hunk 2]
```diff
@@ -18367,7 +18367,8 @@ void skill_enchant_elemental_end(struct block_list *bl, int type)
...
+		status_change_end(bl, SC_ENCHANTARMS, INVALID_TIMER); // Should always end except on equip change
```
Comment (Lemongrass3110) [inline src/map/skill.cpp:18371]:
```suggestion
		status_change_end(bl, SC_ENCHANTARMS, INVALID_TIMER); // Should always end except on equip change
	// If it is on equip change
	else{
		// Check for seven wind (but not level seven!)
		if( sc->data[SC_SEVENWIND] && sc->data[SC_SEVENWIND]->val1 < 7 ){
			return;
		}
	}
```

**Review by Lemongrass3110 — APPROVED** [review]
Please also adjust the call to this function in pc.cpp::pc_unequipitem and remove the condition I merged into this function with my suggestion as we discussed on Discord.

**Review by Lemongrass3110 — APPROVED** [review]

**Comment by ppardo23** [comment]
Hello everyone. I edited src/map/skill.cpp and src/map/pc.cpp as indicated and the problem persists. Using an assassin cross, I switch the dagger from my left hand and lose the elemental converter buff. please help

**Pull request merged**